gammapy/estimators/map_profile.py

```diff
@@ -0,0 +1,313 @@
+            the list of results (list of keys: x_min, x_ref, x_max, alpha, counts, background, excess, ts, sqrt_ts, err,
+             errn, errp, ul, exposure, solid_angle)
+        """
+        if steps == "all":
+            steps = ["err", "ts", "errn-errp", "ul"]
+
+        results = []
+        for index, reg in enumerate(self.regions):
+            spds = dataset.to_spectrum_dataset(reg)
+            mask = spds.mask if spds.mask is not None else slice(None)
+            if isinstance(spds, SpectrumDatasetOnOff):
+                stats = WStatCountsStatistic(
+                    spds.counts.data[mask].sum(),
+                    spds.counts_off.data[mask].sum(),
+                    spds.alpha.data[0,0,0] # At some point, should replace with averaging over energy
+                )
+            else:
+                stats = CashCountsStatistic(
+                    spds.counts.data[mask].sum(),
+                    spds.background.data[mask].sum(),
+                )
+
+            result = {
+                "x_min": self.axis.edges[index],
+                "x_max": self.axis.edges[index+1],
+                "x_ref": self.axis.center[index]
+            }
+            if isinstance(spds, SpectrumDatasetOnOff):
+                result["alpha"] = stats.alpha
+            result.update({
+                "counts": stats.n_on,
+                "background": stats.background,
+                "excess": stats.excess,
+            })
+
+            if "ts" in steps:
+                if isinstance(stats.delta_ts, list) or isinstance(stats.delta_ts, np.ndarray):
+                    result["ts"] = np.asanyarray(stats.delta_ts[0],dtype=np.float32)
+                else:
+                    result["ts"] = np.asanyarray(stats.delta_ts, dtype=np.float32)
+                if isinstance(stats.significance, list) or isinstance(stats.significance, np.ndarray):
+                    result["sqrt_ts"] = np.asanyarray(stats.significance[0], dtype=np.float32)
+                else:
+                    result["sqrt_ts"] = np.asanyarray(stats.significance, dtype=np.float32)
+
+            if "err" in steps:
+                result["err"] = np.asanyarray(stats.error, dtype=np.float32)
+
+            if "errn-errp" in steps:
+                result["errn"] = stats.compute_errn(self.n_sigma)
+                #ToDo: check the type of errp compared to errn (maybe not the same)
+                _errp = stats.compute_errp(self.n_sigma)
+                if isinstance(_errp, list) or isinstance(_errp, np.ndarray):
+                    result["errp"] = np.asanyarray(_errp[0], dtype=np.float32)
+                else:
+                    result["errp"] = np.asanyarray(_errp, dtype=np.float32)
+ 
+            if "ul" in steps:
+                result["ul"] = stats.compute_upper_limit(self.n_sigma_ul)
+ 
+            # What exposure to take? And waht about the mask?
+            result.update(
+                {"exposure": spds.aeff.data.data.max() * spds.livetime,
```

Comment:
I agree the exposure is not well defined here, I would actually remove it...

Reply:
Really, I would like not! The whole purpose of using a SpectrumDataset is to be able to have flux profile. And I rewrote the old class with this objective.
When in the future, the function to_spectrum_dataset will offer options to compute the IRFs in a different manner (e.g. mean over a region, center of a region, etc), then we will fold these options in this estimator.

However, your comment reminds me that I forget to mention an important point about the standard use case of usage of this estimator. We run it after the Ring (ie 1 energy bin) and we make a profile. In this case, the computation of the exposure is correct as the method to_spectrum_dataset compute correctly the effective area (the average exposure divided by time).
In case of usage after the FoVMaker, then we can have several energy bins. And in this case, one can have issues!

So, to conclude, I propose to precise in the documentation to run this estimator with maps produced by the ring. Does this solution acceptable?

Reply:
Thanks @bkhelifi, for the additional explanation. I think using the `ProfileEstimator` with the ring background only, would be a huge limitation. I'm pretty sure users want to use it with data cubes as well.

I fully agree the `ProfileEstimator` should compute a flux profile as well, but we should do it correctly and take into account the energy dependence of the exposure. So compute the flux internally per energy bin and the sum up. As the energy dependence of the exposure is lost in the final result table, I was in favour removing it, but the result should definitely contain a flux column!

Reply:
The last version contains a major upgrade, introducing the management of energy bins and also the flux computation by the dataset. I 'resolve' thus this comment as it does not stand anymore.